pkg/importer/avro/tests/complex_test.arrai

```diff
@@ -0,0 +1,135 @@
+//test.assert.equal(
+`##########################################
+##                                      ##
+##  AUTOGENERATED CODE -- DO NOT EDIT!  ##
+##                                      ##
+##########################################
+Avro[avro_spec="1.0", package="com.example"]:
+    # Types
+
+    !type EventUUID:
+        id <: string
+
+
+    !type Event[namespace="com.example"]:
+        action <: Action
+        header <: Header
+        body <: Party?[union=['Party', 'null']]
+        mapRecords <: set of StringMapRecordItem
+        relationships <: sequence of Relationship
+        tagList <: sequence of string[default="['tag1', 'tag2']"]
+        len <: md5
+        width <: width
+        occurrenceDateTime <: datetime[~timestamp-millis, description="The date and time, to millisecond precision, that this event occurred"]
+        amount <: decimal(4.2)[description="Amount", default="0"]
+
+
+    !type Header:
+        eventUUID <: EventUUID
+
+
+    !type Owner:
+        id <: string
+        ownerTags <: sequence of OwnerTag?
+
+
+    !type OwnerTag:
+        id <: string
+        tag <: string
+
+
+    !type Party[description="Party Details"]:
+        id <: string[description="A unique identifier for party"]
+        subject <: Subject
+
+
+    !type Relationship[namespace="com.example", description="account relationship."]:
+        owner <: Owner?
+        performedAction <: Action
+        relationshipType <: string[description="This would be the business key for uniquely identifying a relation"]
+        startDate <: string?[description="Relationship start date", default="null"]
+        endDate <: string?[union=['string', 'null'], description="Relationship end date", default="null"]
+        issueDate <: IntString?[union=['int', 'string', 'null'], description="Relationship issue date", default="null"]
+        closeDate <: StringInt?[description="Relationship close date", default="null"]
+        realCloseDate <: StringInt?[union=['string', 'null', 'int'], description="Relationship real close date", default="null"]
+        updateDate <: StringInt[description="Relationship update date", default="null", order="ascending"]
+        updateDate1 <: StringInt[description="Relationship update date", default="null", order="ascending", alias_of="updateDate"]
+        updateDate2 <: StringInt[description="Relationship update date", default="null", order="ascending", alias_of="updateDate"]
+        titleList <: sequence of string?[default="[]"]
+        tagList <: StringSequenceTagListItemSequence?[union=['StringSequence', 'null', 'tagListItemSequence'], default="[]"]
+        titleMap <: set of string
+        tagMap <: StringSetTagMapItemSet?
+
+    !alias P1:
+        Relationship
+
+    !alias P2:
+        Relationship
+    !type StringMapRecordItem[json_map_key="key"]:
+        key <: string
+        value <: MapRecord
+
+
+    !type MapRecord:
+        id <: string
+
+
+    !type StringtagMapItemItem[json_map_key="key"]:
```

Comment:
Nit: if this is the name pattern we're going for, capitalise the start of each part (i.e. the T in TagMapItem). Alternatively something like String_tagMapItem_Item
 ```suggestion
    !type StringTagMapItemItem[json_map_key="key"]:
```